### ISSUE TITLE
Search feature

### ISSUE THREAD

**Issue by sharchun** [sameerdash2/comment-viewer#13]
The ability to search / filter all comments for specific text strings would be a great feature.  Especially seeing as youtube doesn't allow this ability, it would really give a big advantage to using your script.  Thx for all you do!

**Comment by sharchun** [comment]
Been using this a lot and I've thought of a couple more suggestions that would push this over the top to be a fantastic script.

1. Search comment contents as mentioned above
2. Get video URL from the query string.  For example "https://commentviewer.com/?v=xxxxxxxxxxx" automatically loads comments from the video with that ID
3. Generate an RSS feed to be notified of new comments on a video

Maybe not all possible to do but take these suggestions for what they're worth.

**Comment by sameerdash2** [comment]
> The ability to search / filter all comments for specific text strings would be a great feature.

I agree it'd be a great feature, and I even implemented it back in 2020 ([see here](https://github.com/sameerdash2/comment-viewer/blob/search/pics/search.png)). But after a couple months the search queries became really slow, taking 10-30 seconds and blocking the entire site for everybody in the process. I then tried switching from sqlite's FTS4 engine to FTS5, which just started throwing errors on 99% of search queries. So eventually I just disabled the whole thing.

Long story short: I don't know how to make full-text search work with SQLite, and I'm kind of retired from adding new features to this project. (unless a lot of people want it 🤷) The old search code is still tagged in this repo's history if any coders want to play around with it -- https://github.com/sameerdash2/comment-viewer/blob/search/src/database.js

> Get video URL from the query string. For example "https://commentviewer.com/?v=xxxxxxxxxxx" 

Unless i'm mistaken, that feature already exists exactly as you described it? 😂 https://github.com/sameerdash2/comment-viewer/issues/5

https://commentviewer.com/?v=vkAezOlnwtk

> Generate an RSS feed to be notified of new comments on a video

I didn't know people still used RSS, lol. But like the above, that'd need a lot of new work and only a few people would use it. Interesting idea though.

**Comment by sharchun** [comment]
OK thanks for the reply.  I might play around with the code myself in future, didn't realize it was a semi retired project.  Thanks for making it.
